src/sql/workbench/contrib/notebook/test/browser/notebookEditor.test.ts

```diff
@@ -155,19 +213,234 @@ suite('Test class NotebookEditor', () => {
 	});
 
-	// NotebookEditor-getCellEditor tests.
-	['', undefined, null, 'unknown string', /*unknown guid*/generateUuid()].forEach(input => {
-		test(`NotebookEditor: Negative Test for getCellEditor() - returns undefined for invalid or unknown guid:'${input}'`, async () => {
-			const result = notebookEditor.getCellEditor(input);
-			assert.strictEqual(result, undefined, `notebookEditor.getCellEditor() should return undefined when invalid guid:'${input}' is passed in for a notebookEditor of an empty document.`);
+	for (const input of ['', undefined, null, 'unknown string', /*unknown guid*/generateUuid()]) {
+		test(`NotebookEditor: getCellEditor() returns undefined for invalid or unknown guid:'${input}'`, async () => {
+			await setupNotebookEditor(notebookEditor, untitledNotebookInput);
+			const inputGuid = <string>input;
+			const result = notebookEditor.getCellEditor(inputGuid);
+			assert.strictEqual(result, undefined, `notebookEditor.getCellEditor() should return undefined when invalid guid:'${inputGuid}' is passed in for a notebookEditor of an empty document.`);
 		});
-	});
+	}
 
-	test('NotebookEditor: Positive Test for getCellEditor() - returns a valid text editor object for valid guid input', async () => {
+	test('NotebookEditor: getCellEditor() returns a valid text editor object for valid guid input', async () => {
+		await setupNotebookEditor(notebookEditor, untitledNotebookInput);
 		const result = notebookEditor.getCellEditor(cellTextEditorGuid);
 		assert.strictEqual(result, queryTextEditor, 'notebookEditor.getCellEditor() should return an expected QueryTextEditor when a guid corresponding to that editor is passed in.');
 
 	});
 
+	test('NotebookEditor: Verifies that domNode passed in via addOverlayWidget() call gets attached to the root HtmlElement of notebookEditor', async () => {
+		await setupNotebookEditor(notebookEditor, untitledNotebookInput);
+		const domNode: HTMLElement = document.createElement('div');
+		const widget: IOverlayWidget = {
+			getId(): string { return ''; },
+			getDomNode(): HTMLElement { return domNode; },
+			getPosition(): IOverlayWidgetPosition | null { return null; }
+		};
+		notebookEditor.addOverlayWidget(widget);
+		const rootElement: HTMLElement = notebookEditor['_overlay'];
+		assert.ok(domNode.parentElement === rootElement, `parent of the passed in domNode must be the root element of notebookEditor:${notebookEditor}`);
+	});
+
+	test('NotebookEditor: Noop methods do not throw', async () => {
+		assert.doesNotThrow(() => {
+			notebookEditor.focus();
+			notebookEditor.layoutOverlayWidget(undefined);
+			assert.strictEqual(notebookEditor.deltaDecorations(undefined, undefined), undefined, 'deltaDecorations is not implemented and returns undefined');
+		});
+	});
+
+
+	test('NotebookEditor: Tests that getConfiguration returns the information set by layout() call', async () => {
```

Comment:
Instead of putting NotebookEditor: before each of these just wrap them all in a suite with the title `NotebookEditor` and mocha will take care of displaying them nicely.

Reply:
done